compute_tools/src/http/api.rs

```diff
@@ -121,8 +123,36 @@ async fn routes(req: Request<Body>, compute: &Arc<ComputeNode>) -> Response<Body
             }
         }
 
+        // download extension files from S3 on demand
+        (&Method::POST, route) if route.starts_with("/extension_server/") => {
+            info!("serving {:?} POST request", route);
+
+            let filename = route.split('/').last().unwrap();
+
+            info!(
+                "serving /extension_server POST request, filename: {:?}",
+                filename
+            );
+
+            match extension_server::download_extension(
+                &compute.ext_remote_storage,
+                ExtensionType::Shared,
+                &compute.pgbin,
+            )
+            .await
+            {
+                Ok(_) => Response::new(Body::from("OK")),
+                Err(e) => {
+                    error!("download_file failed: {}", e);
```

Comment:
NIT: It says `download_file failed`, but I don't see `download_file` in the whole PR

Reply:
Yeah, it should say `download_extension`. Updated it now.